manifests/init.pp

```diff
@@ -2,10 +2,14 @@
 #
 # @param config
 #   The contents of the configuration file, if any
+# @param manage_package_repo Installs the package repositories
 class grafana_alloy (
-  Optional[String[1]] $config = undef,
+  Optional[String[1]] $config  = undef,
+  Boolean $manage_package_repo = true,
 ) {
-  contain grafana_alloy::repo
+  if $grafana_alloy::manage_package_repo {
```

Comment:
I think you need to move `Class['grafana_alloy::repo'] -> Class['grafana_alloy::install']` inside this `if` block